### PR TITLE
upgrade eks module

### PR DESCRIPTION
**Pull request by smalltown**
- upgrade eks module for supporting latest version
- delete worker-asg, worker-spot, worker-common module, creating eks-worker module by AWS launch template to support spot/on-demand EC2 instance type

### CODE REVIEW

File: modules/aws/eks/variables.tf [hunk 1]
```diff
@@ -0,0 +1,169 @@
+variable "aws_region" {
+  description = "The AWS region to build eks cluster"
+  type        = "string"
+  default     = ""
+}
+
+variable "cidr_access_eks_endpoint" {
```
Comment (owensengoku) [inline modules/aws/eks/variables.tf]: How about to rename `cidr_access_eks_https` for consistency with `cidr_access_worker_ssh`?

File: modules/aws/eks/variables.tf [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+  description = "(Optional) Desired Kubernetes master version. If you do not specify a value, the latest available version is used."
+}
+
+variable "config_output_path" {
+  description = "The path to store config, e.g. kubeconfig"
+  type        = "string"
+  default     = "./terraform"
+}
+
+variable "write_kubeconfig" {
```
Comment (owensengoku) [inline modules/aws/eks/variables.tf]: rename to `kubeconfig_output_flag` ?

File: modules/aws/eks/variables.tf [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+  description = "Whether to write a Kubectl config file containing the cluster configuration. Saved to `config_output_path`."
+  default     = true
+}
+
+variable "manage_aws_auth" {
+  description = "Whether to apply the aws-auth configmap file."
+  default     = true
+}
+
+variable "write_aws_auth_config" {
```
Comment (owensengoku) [inline modules/aws/eks/variables.tf]: rename to aws_auth_config_output_flag ?

File: modules/ignitions/eks-kube-config/kubeconfig.tf [hunk 4]
```diff
@@ -0,0 +1,43 @@
+locals {
+  kubeconfig_name = "${var.kubeconfig_name == "" ? "${var.cluster_name}" : var.kubeconfig_name}"
```
Comment (owensengoku) [inline modules/ignitions/eks-kube-config/kubeconfig.tf]: Consider to give a default name ?

File: modules/ignitions/eks-kube-config/resources/kubernetes/kubeconfig [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+clusters:
+- cluster:
+    server: ${cluster_endpoint}
+    certificate-authority-data: ${cluster_auth_base64}
+  name: ${kubeconfig_name}
+
+contexts:
+- context:
+    cluster: ${kubeconfig_name}
```
Comment (owensengoku) [inline modules/ignitions/eks-kube-config/resources/kubernetes/kubeconfig]: Of course, it does not matter to use the same name, but I guess could consider to use ${cluster_name} here.

File: modules/aws/kube-worker/ami.tf [hunk 6]
```diff
@@ -0,0 +1,39 @@
+locals {
+  ami_owner = "595879546273"
```
[the remark this answers is not in the document]
Comment (smalltown) [reply modules/aws/kube-worker/ami.tf]: no problem

File: modules/aws/eks/variables.tf (hunk 1 above)
Comment (smalltown) [reply modules/aws/eks/variables.tf]: sure

File: modules/aws/eks/variables.tf (hunk 2 above)
Comment (smalltown) [reply modules/aws/eks/variables.tf]: OK!

File: modules/aws/eks/variables.tf (hunk 3 above)
Comment (smalltown) [reply modules/aws/eks/variables.tf]: yes

**Pull request merged**